docs/checklist-command.md

```diff
@@ -0,0 +1,32 @@
+# New CLI command
+
+Subcommands should always be preferred over adding more top-level commands.
+
+Code flow for commands is generally:
+
+```
+CLI (command/) -> API Client (api/) -> HTTP API (command/agent) -> RPC (nomad/)
+```
+
+## Code
+
+1.  [ ] New file in `command/` or in an existing file if a subcommand
+2.  [ ] Test new command in `command/` package
+3.  [ ] Implement autocomplete
+4.  [ ] Implement `-json` (returns raw API response)
+5.  [ ] Implement `-verbose` (expands truncated UUIDs, adds other detail)
+7.  [ ] Update help text
+7.  [ ] Implement and test new HTTP endpoint in `command/agent/<command>_endpoint.go`
+8.  [ ] Implement and test new RPC endpoint in `nomad/<command>_endpoint.go`
+9.  [ ] Implement and test new Client RPC endpoint in `client/<command>_endpoint.go`
+  * For client endpoints only (e.g. Filesystem)
```

Comment:
double checking that you mean "only do steps 10-12 for client endpoints (e.g. fs)"

Reply:
![Screenshot from 2019-06-06 12-43-27](https://user-images.githubusercontent.com/1182129/59061683-df844d00-8858-11e9-8c34-62d187ef59fc.png)

Reply:
Hm, good question. I meant it for Step 9 only and made it a sub-bullet-point, but it renders ambiguously:

> 8.  [ ] Implement and test new RPC endpoint in `nomad/<command>_endpoint.go`
> 9.  [ ] Implement and test new Client RPC endpoint in `client/<command>_endpoint.go`
>  * For client endpoints only (e.g. Filesystem)
> 10. [ ] Implement and test new `api/` package Request and Response structs
> 11. [ ] Implement and test new `api/` package helper methods

I'll just put it in Step 9 directly for now to make it clear.